cpp/arcticdb/storage/s3/s3_storage.hpp

```diff
@@ -235,8 +235,12 @@ auto get_s3_config(const ConfigType& conf) {
     client_configuration.maxConnections = conf.max_connections() == 0 ?
             ConfigsMap::instance()->get_int("VersionStore.NumIOThreads", 16) :
             conf.max_connections();
-    client_configuration.connectTimeoutMs = conf.connect_timeout() == 0 ? 30000 : conf.connect_timeout();
-    client_configuration.requestTimeoutMs = conf.request_timeout() == 0 ? 200000 : conf.request_timeout();
+    client_configuration.connectTimeoutMs = conf.connect_timeout() == 0 ?
+                                          ConfigsMap::instance()->get_int("S3Storage.ConnectTimeoutMs", 30000) :
+                                          conf.connect_timeout();
+    client_configuration.httpRequestTimeoutMs = ConfigsMap::instance()->get_int("S3Storage.HttpRequestTimeoutMs", 0);
```

Comment:
I don't know if there was ever a way to set `conf.request_timeout()` on the library config but now we don't look at it at all.

Also for the connection timeout it feels like a better approach is to allow configsmap to override anything written in the library config (which is hard to change), i.e.
`ConfigsMap::instance()->get_int("S3Storage.ConnectTimeoutMs", conf.connect_timeout() == 0 ? 3000 : conf.connect_timeout())`